gridmet.py

```diff
@@ -0,0 +1,152 @@
+
+# Variables
+gdf_nhru02_path = './data/nhru_02/nhru_02.shp'
+start_date = '1979-01-01'
+end_date = '1989-01-01'
+lon_min = -79
+lon_max = -67
+lat_min = 36
+lat_max = 42
+output_path = './data/'
+## official list of variables needed for drb-inland-salinity model
+data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs','rmax','rmin','sph']
+data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+         'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+         'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+         'daily_minimum_relative_humidity','daily_mean_specific_humidity']
+
+## Shorter list of data variables for testing:
+data_vars_grd2shp_shrt = ['tmmn', 'vs']
+
+## Full function
+def g2shp_regridding(polygon_file_path,
+                     output_data_folder,
+                     var_short,
+                     start_date, end_date,
+                     lon_min, lon_max, lat_min, lat_max,
+                     g2s_file_prefix = 'tmp'):
+    """
+    :param str polygon_file_path: path to hru shapefile used for regridding of gridMET data
+    :param str output_data_folder: folder location where outputs such as weights pickle file and netcdf output file will reside
+    :param str/list var_short: data variable short name or list of data variable short names. Must be one of the following: ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    :param str start_date: Start date of data collection yyyy-mm-dd
+    :param str end_date: End date of data collection yyyy-mm-dd
+    :param str lon_min: bbox of aoi longitude min
+    :param str lon_max: bbox of aoi longitude max
+    :param str lat_min: bbox of aoi latitude min
+    :param str lat_max: bbox of aoi latitude max
+    :param str g2s_file_prefix: prefix used to save the grd2shp_xarray
+    :return: g2s array
+    """
+
+    ## Create dictionary of short variable names with official long variable name
+    data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+                          'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+                          'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+                          'daily_minimum_relative_humidity', 'daily_mean_specific_humidity']
+
+    all_data_vars_dict = dict(zip(data_vars_shrt_all, data_vars_long_all))
+    subset = {key: all_data_vars_dict[key] for key in var_short}
+    print(subset)
+    del all_data_vars_dict
+
+    ## geospatial vector file
+    gdf = gpd.read_file(polygon_file_path)
+
+    ## check if only 1 var (short and long) is provided, change to list
+    if type(var_short) == str:
+        var_short = [var_short]
+        print(var_short)
+    ## check that var_short is a list
+    elif type(var_short) != list:
+        var_short = [var_short]
+        print(var_short)
+
```

Comment:
I think it's a little cleaner to use `isinstance()`. Also, wouldn't the second condition (`elif type(var_short) != list`) also be True if the first condition were true? I think we could just have the second condition:
```suggestion
    if not isinstance(var_short, list):
        var_short = [var_short]
        print(var_short)

```

Reply:
Agreed and thanks for the tip. I think on the moment, i thought their could be  an instance where multiple var_short inputs could be passed as a different type than a list. This is.instance() check works much better.